### PR TITLE
Switch from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by neomilium**
ATM, this PR drops the IRC notification feature.

Tell me if you still want it.

### CODE REVIEW

**Comment by neomilium** [comment]
Converting to draft and waiting on PR #202 to be merged, as checks failed for an unknown reason on top of `master` but runs well on top of `improve-tests`.

https://github.com/opus-codium/modulesync/pull/1

**Comment by neomilium** [comment]
Before merging this PR, maintainers need to add secret keys for releasing process (i.e. in `.github/workflows/release.yml`).

Remind: The IRC notification feature is dropped.

Please consider this contribution as a base, not a fully tested code, multiple reviews appreciated.

File: .github/workflows/ci.yml
Comment (ekohl) [inline .github/workflows/ci.yml:1]: It looks like you don't have a name for this and the filename isn't great.

**Review by ekohl — CHANGES_REQUESTED** [review]
I pushed the branch to upstream which triggers the tests.

File: .github/workflows/ci.yml
Comment (bastelfreak) [reply .github/workflows/ci.yml:1]: @neomilium can you take a look at this?

Comment (neomilium) [reply .github/workflows/ci.yml:1]: It seems I will not have so much spare time during next weeks. So, feel free to rework my PR if you want to merge it quickly.

Comment (neomilium) [reply .github/workflows/ci.yml:1]: @ekohl What name you prefer `ci`, `Continuous integration`, `Tests`? Or maybe you have better suggestion. Thx

Comment (ekohl) [reply .github/workflows/ci.yml:1]: I'm leaning to keeping it short and use `CI`.

Comment (neomilium) [reply .github/workflows/ci.yml:1]: OK, so `CI` for the name but you point me that `ci.yml` was not a good filename. What did you expect here? ;-)

Comment (ekohl) [reply .github/workflows/ci.yml:3]: I can't link back to anymore, but if you go to Checks, you see this:

image

Previously that showed the filename instead of `CI`. Looking at it now, Continuous integration would probably also have been fine since there's sufficient space. I hope that helps to understand why I cared about it.

**Review by ekohl — APPROVED** [review]
Thanks!

File: .github/workflows/ci.yml
Comment (neomilium) [reply .github/workflows/ci.yml:3]: OK, thx for explanations.

**Pull request merged**
